pkgs/by-name/cl/clprover/package.nix

```
@@ -0,0 +1,66 @@
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+}:
+
+stdenv.mkDerivation (self: {
+  pname = "CLProver++";
+  version = "1.0.3";
+
+  src = fetchFromGitHub {
+    owner = "PaulGainer";
+    repo = "CLProverPlusPlus";
+    rev = "66bf6dc19d77094cc06eac3f30a3630ba830ac86";
+    sha256 = "sha256-UZ5e11wGuKyVrG+7hZJY6OmN6Y1mg43xPuvXPRVNKNw=";
+  };
+
+  configurePhase = ''
+    runHook preConfigure
+
+    sed -i 's/\(TARGET_OS *:= *\)[^ ]+/\1${
+      if stdenv.hostPlatform.isWindows then "WINDOWS" else "LINUX"
+    }/g' Makefile
+    sed -i 's/-m64/${
+      if stdenv.hostPlatform.isArmv7 then
+        "-march=armv7"
+      else if stdenv.hostPlatform.isAarch64 then
+        "-march=armv8"
+      else if stdenv.hostPlatform.is32bit then
+        "-m32"
+      else
+        "-m64"
+    }/g' Makefile
+
+    runHook postConfigure
+  '';
+
+  preBuild = ''
+    mkdir bin
+  '';
+
+  makeFlags = [
+    "CC=${lib.getBin stdenv.cc}/bin/${stdenv.cc.targetPrefix}g++"
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    install -m755 bin/${self.meta.mainProgram} $out/bin/${self.meta.mainProgram}
+
+    mkdir -p $out/share/${self.pname}
+    cp -r examples $out/share/${self.pname}/examples
+
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "Ordered resolution based theorem prover for Coalition Logic";
+    homepage = "https://cgi.csc.liv.ac.uk/~ullrich/CLProver++/";
+    maintainers = with lib.maintainers; [ mgttlinger ];
+    platforms = with lib.platforms; linux ++ windows;
```

Nit: This is the more commonly used formt

```suggestion
    platforms = with lib.platforms; [
      linux
      windows
    ];
```

---

That doesn't type as both `linux` and `windows` are lists of platform strings.

---

Please disregard and mark as resolved